modules/luci-mod-network/htdocs/luci-static/resources/tools/network.js

```diff
@@ -519,7 +520,395 @@ return baseclass.extend({
+		};
+		o.depends('type', 'bonding');
+
+		o = this.replaceOption(s, 'devgeneral', form.ListValue, 'policy', _('Bonding Policy'));
+		o.default = 'active-backup';
+		o.value('active-backup', _('Active backup'));
+		o.value('balance-rr', _('Round robin'));
+		o.value('balance-xor', _('Transmit hash - balance-xor'));
+		o.value('broadcast', _('Broadcast'));
+		o.value('802.3ad', _('LACP - 802.3ad'));
+		o.value('balance-tlb', _('Adaptive transmit load balancing'));
+		o.value('balance-alb', _('Adaptive load balancing'));
+		o.cfgvalue = function(/* ... */) {
+			var val = form.ListValue.prototype.cfgvalue.apply(this, arguments);
+
+			switch (val || '') {
+			case 'active-backup':
+			case '0':
+				return 'active-backup';
+
+			case 'balance-rr':
+			case '1':
+				return 'balance-rr';
+
+			case 'balance-xor':
+			case '2':
+				return 'balance-xor';
+
+			case 'broadcast':
+			case '3':
+				return 'broadcast';
+
+			case '802.3ad':
+			case '4':
+				return '802.3ad';
+
+			case 'balance-tlb':
+			case '5':
+				return 'balance-tlb';
+
+			case 'balance-alb':
+			case '6':
+				return 'balance-alb';
+
+			default:
+				return 'active-backup';
+			}
+		};
+		o.depends('type', 'bonding');
+
+		o = this.replaceOption(s, 'devgeneral', form.Flag, 'all_ports_active', _('All ports active'), _('Allow receiving on inactive ports'));
+		o.default = o.disabled;
+		o.depends('type', 'bonding');
+
+		o = this.replaceOption(s, 'devgeneral', widgets.DeviceSelect, '_net_device_primary', _('Primary Device'));
+		o.ucioption = 'primary';
+		o.rmempty = true;
+		o.noaliases = true;
+		o.nobridges = true;
+		o.optional = false;
+		o.depends({'type': 'bonding', 'policy': 'active-backup'});
+
+		o = this.replaceOption(s, 'devadvanced', form.ListValue, 'xmit_hash_policy', _('Slave selection hash policy'));
+		o.default = '';
+		o.value('', _(''));
+		o.value('layer2', _('Layer 2'));
+		o.value('layer2+3', _('Layer 2+3'));
+		o.value('layer3+4', _('Layer 3+4'));
+		o.value('encap2+3', _('Encap 2+3'));
+		o.value('encap3+4', _('Encap 3+4'));
+		o.cfgvalue = function(/* ... */) {
+			var val = form.ListValue.prototype.cfgvalue.apply(this, arguments);
+
+			switch (val || '') {
+			case 'layer2':
+			case '0':
+				return 'layer2';
+
+			case 'layer2+3':
+			case '1':
+				return 'layer2+3';
+
+			case 'layer3+4':
+			case '2':
+				return 'layer3+4';
+
+			case 'encap2+3':
+			case '4':
+				return 'encap2+3';
+
+			case 'encap3+4':
+			case '5':
+				return 'encap3+4';
+
+			default:
+				return '';
+			}
+		};
+		o.depends({'type': 'bonding', 'policy': 'balance-xor'});
+		o.depends({'type': 'bonding', 'policy': '802.3ad'});
+		o.depends({'type': 'bonding', 'policy': 'balance-tlb'});
+
+		o = this.replaceOption(s, 'devadvanced', form.Value, 'ad_actor_system', _('MAC address for LACPDUs'));
+		o.description = _('This specifies the mac-address for the actor in protocol packet exchanges (LACPDUs). The value cannot be NULL or multicast.');
+		o.datatype = 'macaddr';
```

Comment:
Perhaps a couple of error checks here to prevent NULL and multicast.

Can we add any predefined values here? e.g. `01:80:C2:XX:XX:XX`? Or should this be the underlying MAC of the device(s) - which one could also probe for and provide?

Reply:
Added to my TO-DO list